agent/agent.go

```diff
@@ -253,16 +260,24 @@ func (a *Agent) UpdatePeers(ctx context.Context, p pool.Pool) error {
 
 // AddPeers requests num peers from the pool and connects the node to them.
 func (a *Agent) AddPeers(ctx context.Context, p pool.Pool, num int) error {
-	logger.Printf("Requesting %d more %q hosts from pool...", a.EthNode.Kind(), num)
+	kind := "" // Any kind of node by default
+	if !a.nodeInfo.IsFullNode {
+		// Non-full-nodes need nodes of the same kind to peer with (light nodes)
+		kind = a.nodeInfo.Kind.String()
+	}
+
+	logger.Printf("Requesting more kind=%q peers from pool: %d", kind, num)
 	peerResp, err := p.Peer(ctx, pool.PeerRequest{
 		Num:  num,
-		Kind: a.nodeInfo.Kind.String(),
+		Kind: kind,
 	})
-	if err != nil {
+	if err != nil && jsonrpc2.IsErrorCode(err, jsonrpc2.ErrCodeInternal) && strings.HasPrefix(err.Error(), "no available") {
```

Comment:
Borderline silly. Maybe it shouldn't be a JSONRPC error? Though feels like it should be an error if you request more peers to be whitelisted and none were.

Reply:
Maybe add some sort of helper that converts a Golang `error` to a jsonrpc2 Error struct?  Naming below isn't ideal:

```golang
if e := jsonrpc2.ErrAsError(err); e != nil {
  if e.Code == jsonrpc2.ErrCodeInternal && e.Message.HasPrefix("no available") {
   return ErrNoPeers
  }
}
```

Reply:
Looks too similar to what we already have I think. Will sleep on this one.